azkaban-common/src/main/java/azkaban/project/FlowLoaderUtils.java

```diff
@@ -38,6 +46,81 @@
 
   private static final Logger logger = LoggerFactory.getLogger(FlowLoaderUtils.class);
 
+  /**
+   * Sets props in flow yaml file.
+   *
+   * @param path the flow or job path delimited by ":", e.g. "flow:subflow1:subflow2:job3"
+   * @param flowFile the flow yaml file
+   * @param prop the props to set
+   */
+  public static void setPropsInYamlFile(final String path, final File flowFile, final Props prop) {
```

Comment:
Will the method be called by multiple threads? if so is it thread safe?

Reply:
Good question! Yes it will be called by multiple threads when overriding job properties. If two threads are writing flow files into DB with the same primary key (project id, project version, flow name, flow version), then the second thread would fail in DB insertion. This is better than adding synchronization, which is blocking and could be slow. It makes sense to explicitly fail the second thread so that the user is aware of the conflict in writing to the same file and could retry again.